src/coreComponents/mesh/generators/PrismUtilities.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_MESH_GENERATORS_PRISMUTILITIES_HPP_
+#define GEOSX_MESH_GENERATORS_PRISMUTILITIES_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Span.hpp"
+
+namespace geosx
+{
+
+/**
+ * @brief Get the local indices of the nodes in a face of the prism with N-sided polygon base.
+ * @tparam N the number of sides in the polygon base
+ * @param[in] faceNum the local index of the target face in the element  (this will be 0-numFacesInElement)
+ * @param[in] elemNodes Element to nodes mapping.
+ * @param[out] faceNodes space to write node indices to, must be of sufficient size
+ * @return number of nodes in the face
+ */
+template< localIndex N >
+localIndex getFaceNodesPrism( localIndex const faceNum,
+                              arraySlice1d< localIndex const, cells::NODE_MAP_USD - 1 > const & elemNodes,
+                              Span< localIndex > const faceNodes )
+{
```

Comment:
Assuming the order of faces is not important. Case (faceNum==0) can be combined with  case ( faceNum >= 3 && faceNum <= N ) -> ( faceNum >= 2 && faceNum <= N ).
In this case ( faceNum == 2 ) becomes ( faceNum == 0 )
For case ( faceNum == 1 ) why not just write:
    for( localIndex i = 0; i <  N; ++i )
    {
      faceNodes[i] = elemNodes[N-i-1];
    }
Of course there are maybe some numbering constraints I am missing.

Reply:
I understand the suggestion. See my reply above https://github.com/GEOSX/GEOSX/pull/1902#discussion_r866171474